src/compose.ts

```diff
@@ -65,15 +66,49 @@ services:
  * does not have the expected structure and discrepancies can't be resolved,
  * validation errors are thrown. The input composition is mutated in-place.
  *
- * @param c The input composition as a plain JS object
+ * @param inputCompositionObject The input composition as a plain JS object
  */
-export function normalize(o: any): Composition {
-	if (!_.isObject(o)) {
+export function normalize(inputCompositionObject: any): Composition;
+
+/**
+ * Validates, normalises and returns the input composition. If the composition
+ * does not have the expected structure and discrepancies can't be resolved,
+ * validation errors are thrown. The input composition is mutated in-place.
+ *
+ * Expands the composition context with the passed fileResolverCb.
+ * Drops env_file propertie from composition to indicate that the expand has taken place.
```

Comment:
I think these two lines fail to provide any insight into why I’d want as a user to pass the callback and how it’s going to be used or how it should behave.

You also seem to be using terms throughout the file I’m personally not familiar with (which is okay, I just want to understand). What is the “composition context” and what does “Expands the composition context with the passed fileResolverCb” mean?

Reply:
I'll add my understanding of the context to the doc.
My understanding of a context is the project directory which contains the docker-compose file and describes the context for the composition. This is maybe too broad but I've the feeling that it describes the idea best.